drupal-7.41-states-toooooo-slow.patch

```diff
@@ -0,0 +1,134 @@
++        this.element.triggerHandler({ type: 'state:' + this.state, value: value, trigger: true });
++      }
++      else {
++        var event = jQuery.Event('state:' + this.state);
++        event.value = value;
++        event.trigger = true;
++        event.preventDefault();
++        event.stopPropagation();
++        this.element.trigger(event);
++      }
++      $(document).trigger({ type: 'state:' + this.state, value: value, trigger: true, states_target: this.element.get(0) });
+     }
+   },
+ 
+@@ -342,14 +372,36 @@ states.Trigger.prototype = {
+       var value = valueFn.call(this.element, e);
+       // Only trigger the event if the value has actually changed.
+       if (oldValue !== value) {
+-        this.element.trigger({ type: 'state:' + this.state, value: value, oldValue: oldValue });
++        if (jqueryversion == 'new') {
++          this.element.triggerHandler({ type: 'state:' + this.state, value: value, oldValue: oldValue });
++        }
++        else {
++          var event = jQuery.Event('state:' + this.state);
++          event.value = value;
++          event.oldValue = oldValue;
++          event.preventDefault();
++          event.stopPropagation();
++          this.element.trigger(event);
++        }
++        $(document).trigger({ type: 'state:' + this.state, value: value, oldValue: oldValue, states_target: this.element.get(0) });
+         oldValue = value;
+       }
+     }, this));
+ 
+     states.postponed.push($.proxy(function () {
+       // Trigger the event once for initialization purposes.
+-      this.element.trigger({ type: 'state:' + this.state, value: oldValue, oldValue: null });
++      if (jqueryversion == 'new') {
++        this.element.triggerHandler({ type: 'state:' + this.state, value: oldValue, oldValue: null });
++      }
++      else {
++        var event = jQuery.Event('state:' + this.state);
++        event.value = oldValue;
++        event.oldValue = null;
++        event.preventDefault();
++        event.stopPropagation();
++        this.element.trigger(event);
++      }
++      $(document).trigger({ type: 'state:' + this.state, value: oldValue, oldValue: null, states_target: this.element.get(0) });
+     }, this));
+   }
+ };
+@@ -480,6 +532,7 @@ $(document).bind('state:disabled', function(e) {
+   // Only act when this change was triggered by a dependency and not by the
+   // element monitoring itself.
+   if (e.trigger) {
++    e.target = e.states_target || e.target;
+     $(e.target)
+       .attr('disabled', e.value)
+         .closest('.form-item, .form-submit, .form-wrapper').toggleClass('form-disabled', e.value)
+@@ -492,6 +492,7 @@ $(document).bind('state:disabled', function(e) {
+ 
+ $(document).bind('state:required', function(e) {
+   if (e.trigger) {
++    e.target = e.states_target || e.target; 
+     if (e.value) {
+       var $label = $(e.target).closest('.form-item, .form-wrapper').find('label');
+       // Avoids duplicate required markers on initialization.
+
```

Comment:
My version of git apply and patch don't like this extra line, because the hunk header isn't valid.